atlp-rwanda/Team-furebo-E-commerce-bn
# Ft admin set roles permissions 184651338

What does this PR do?
This PR adds the functionality for Admin to assign various users to different roles

Description of Task to be completed?
The task was to create a feature that allows Admin to assign different roles to users on the system.
This feature ensures that users can only perform actions allowed by their assigned roles.
The feature also includes the creation of a role for the admin account itself.

How should this be manually tested?

1. Create an Admin account through http://localhost:3000/api/registerAdmin.
2. Login using the credentials through http://localhost:3000/api/login.
3. Obtain the token received upon login.
4. Add the token in the authorization header of the request for endpoint http://localhost:3000/api/updateRole/.
5. Add the ID of the user whose role is to assign in the request path.
6. Add the role parameter in the request body.
7. Send the request.
8. Verify that the user's role has been updated correctly and that they can only perform actions allowed by their assigned roles.

What are the relevant pivotal tracker/Trello stories?
#184651338
